drivers/firmware/tisci/tisci.c

```diff
@@ -0,0 +1,1618 @@
+	req.rxcq_qnum = params->rxcq_qnum;
+	req.rx_priority = params->rx_priority;
+	req.rx_qos = params->rx_qos;
+	req.rx_orderid = params->rx_orderid;
+	req.rx_sched_priority = params->rx_sched_priority;
+	req.flowid_start = params->flowid_start;
+	req.flowid_cnt = params->flowid_cnt;
+	req.rx_pause_on_err = params->rx_pause_on_err;
+	req.rx_atype = params->rx_atype;
+	req.rx_chan_type = params->rx_chan_type;
+	req.rx_ignore_short = params->rx_ignore_short;
+	req.rx_ignore_long = params->rx_ignore_long;
+
+	ret = tisci_do_xfer(dev, xfer);
+	if (ret) {
+		LOG_ERR("UDMAP RX channel %u config failed (ret=%d)", params->index, ret);
+		return ret;
+	}
+
+	LOG_DBG("UDMAP RX channel %u configured successfully", params->index);
+	return 0;
+}
+
+/* PSI-L Management Functions */
+
+int tisci_cmd_rm_psil_pair(const struct device *dev, uint32_t nav_id, uint32_t src_thread,
+			   uint32_t dst_thread)
+{
+	struct tisci_msg_psil_pair_req req;
+	struct tisci_msg_psil_pair_resp resp;
+	struct tisci_xfer *xfer;
+	int ret;
+
+	xfer = tisci_setup_one_xfer(dev, TISCI_MSG_RM_PSIL_PAIR, 0, &req, sizeof(req), &resp,
+				    sizeof(resp));
+	if (!xfer) {
+		LOG_ERR(" Failed to setup PSI-L pair transfer");
+		return -EINVAL;
+	}
+
+	req.nav_id = nav_id;
+	req.src_thread = src_thread;
+	req.dst_thread = dst_thread;
+
+	ret = tisci_do_xfer(dev, xfer);
+	if (ret) {
+		LOG_ERR("PSI-L pair failed nav:%u %u->%u (ret=%d)", nav_id, src_thread, dst_thread,
+			ret);
+		return ret;
+	}
+
+	LOG_DBG("PSI-L pair successful nav:%u %u->%u", nav_id, src_thread, dst_thread);
+	return 0;
+}
+
+int tisci_cmd_rm_psil_unpair(const struct device *dev, uint32_t nav_id, uint32_t src_thread,
+			     uint32_t dst_thread)
+{
+	struct tisci_msg_psil_unpair_req req;
+	struct tisci_msg_psil_unpair_resp resp;
+	struct tisci_xfer *xfer;
+	int ret;
+
+	xfer = tisci_setup_one_xfer(dev, TISCI_MSG_RM_PSIL_UNPAIR, 0, &req, sizeof(req), &resp,
+				    sizeof(resp));
+	if (!xfer) {
+		LOG_ERR(" Failed to setup PSI-L unpair transfer");
+		return -EINVAL;
+	}
+
+	req.nav_id = nav_id;
+	req.src_thread = src_thread;
+	req.dst_thread = dst_thread;
+
+	ret = tisci_do_xfer(dev, xfer);
+	if (ret) {
+		LOG_ERR("PSI-L unpair failed %u->%u (ret=%d)", src_thread, dst_thread, ret);
+		return ret;
+	}
+
+	LOG_DBG("PSI-L unpair successful %u->%u", src_thread, dst_thread);
+	return 0;
+}
+
+/* Interrupt Management Functions */
+static int tisci_cmd_rm_irq_set_raw(const struct device *dev, struct tisci_msg_rm_irq_set_req *req)
+{
+	struct tisci_xfer *xfer;
+	struct tisci_msg_rm_irq_set_resp resp = {0};
+	int ret;
+
+	xfer = tisci_setup_one_xfer(dev, TISCI_MSG_RM_IRQ_SET, 0, req, sizeof(*req), &resp,
+				    sizeof(resp));
+	if (!xfer) {
+		LOG_ERR("Failed to setup transfer");
+		return -EINVAL;
+	}
+
+	ret = tisci_do_xfer(dev, xfer);
+	if (ret) {
+		LOG_ERR("Failed to set IRQ (ret=%d)", ret);
+		return ret;
+	}
+
+	if ((resp.hdr.flags & TISCI_FLAG_RESP_GENERIC_ACK) != TISCI_FLAG_RESP_GENERIC_ACK) {
```

Comment:
The response check is already done as part of `tisci_do_xfer()`, you can drop this check. Same below.